src/lib/util/resolveSwagger.ts

```diff
@@ -293,24 +293,26 @@ export class ResolveSwagger {
     if (!unwrappedProperty) {
       throw new Error("Null unwrapped property.")
     }
+
+    // Note: per https://editor-next.swagger.io/ tooltip when hovering over '$ref',
+    // other properties must be ignored when $ref is present:
+    //
+    //  $ref: string
+    //  Any time a subschema is expected, a schema may instead use an object containing a "$ref" property.
+    //  The value of the $ref is a URI Reference. Resolved against the current URI base,
+    //  it identifies the URI of a schema to use.
+    //  All other properties in a "$ref" object MUST be ignored.
+    //
+    parentProperty = parentProperty.$ref ? this.dereference(parentProperty.$ref) : parentProperty
+    unwrappedProperty = unwrappedProperty.$ref ? this.dereference(unwrappedProperty.$ref) : unwrappedProperty
+
     if ((!parentProperty.type || parentProperty.type === "object") && (!unwrappedProperty.type || unwrappedProperty.type === "object")) {
-      let parentPropertyToCompare = parentProperty
-      let unwrappedPropertyToCompare = unwrappedProperty
-      if (parentProperty.$ref) {
-        parentPropertyToCompare = this.dereference(parentProperty.$ref)
-      }
-      if (unwrappedProperty.$ref) {
-        unwrappedPropertyToCompare = this.dereference(unwrappedProperty.$ref)
-      }
-      if (parentPropertyToCompare === unwrappedPropertyToCompare) {
-        return true
-      }
-      return false
-    }
-    if (parentProperty.type === "array" && unwrappedProperty.type === "array") {
+      return parentProperty === unwrappedProperty
+    } else if (parentProperty.type === "array" && unwrappedProperty.type === "array") {
       return this.isEqual(parentProperty.items, unwrappedProperty.items)
+    } else {
+      return parentProperty.type === unwrappedProperty.type && parentProperty.format === unwrappedProperty.format
     }
-    return parentProperty.type === unwrappedProperty.type && parentProperty.format === unwrappedProperty.format
   }
```

Comment:
This code should be equivalent, I just collapsed L305-308 into 310, and added `else`s for clarity.